XIVLauncher/Updates.cs

```diff
@@ -19,23 +20,28 @@ public async Task Run(bool downloadPrerelease = false)
         {
             // GitHub requires TLS 1.2, we need to hardcode this for Windows 7
             ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
-
-            using (var updateManager = await UpdateManager.GitHubUpdateManager(repoUrl: App.RepoUrl, applicationName: "XIVLauncher", prerelease: downloadPrerelease))
+            try
             {
-                SquirrelAwareApp.HandleEvents(
-                    onInitialInstall: v => updateManager.CreateShortcutForThisExe(),
-                    onAppUpdate: v => updateManager.CreateShortcutForThisExe(),
-                    onAppUninstall: v => updateManager.RemoveShortcutForThisExe());
-
-                var downloadedRelease = await updateManager.UpdateApp();
+                using (var updateManager = await UpdateManager.GitHubUpdateManager(repoUrl: App.RepoUrl, applicationName: "XIVLauncher", prerelease: downloadPrerelease))
+                {
+                    SquirrelAwareApp.HandleEvents(
+                        onInitialInstall: v => updateManager.CreateShortcutForThisExe(),
+                        onAppUpdate: v => updateManager.CreateShortcutForThisExe(),
+                        onAppUninstall: v => updateManager.RemoveShortcutForThisExe());
+                    var downloadedRelease = await updateManager.UpdateApp();
 
-                if (downloadedRelease != null)
-                    UpdateManager.RestartApp();
+                    if (downloadedRelease != null)
+                        UpdateManager.RestartApp();
 #if !XL_NOAUTOUPDATE
-                else
-                    OnUpdateCheckFinished?.Invoke(this, null);
+                    else
+                        OnUpdateCheckFinished?.Invoke(this, null);
 #endif
-
+                }
+            }
+            catch (Exception)
+            {
+                MessageBox.Show("Failed to check for updates. This may be caused by GitHub connectivity issues. Wait a few minutes and try again.", "Update failure", MessageBoxButtons.OK, MessageBoxIcon.Error);
```

Comment:
This needs to be localized(See Loc.Localize) and please adjust the wording to include a reference to the Discord, e.g. "If this doesn't work after several minutes, please report the error on our Discord server."

Reply:
Done.